### ISSUE TITLE
how to set config parameters?

### ISSUE THREAD

**Issue by hotarujun** [bwconrad/vit-finetune#9]
Hello,

  Thank you so much for sharing the code ! 

  I'm training custom data with your code, the dataset is [525 bird images](https://www.kaggle.com/datasets/gpiosenka/100-bird-species). But I don't know how these parameters should be set. Here's my configuration file:

```
trainer:
  accelerator: gpu
  devices: 2
  precision: 16-mixed
  logger:
    class_path: pytorch_lightning.loggers.CSVLogger
    init_args:
      save_dir: /kaggle/working/
      name: custom
  max_steps: 5000
  val_check_interval: 500
model:
  model_name: vit-b16-224-in21k
  training_mode: full
  optimizer: sgd
  lr: 0.01
  momentum: 0.9
  weight_decay: 0.0
  scheduler: cosine
  warmup_steps: 500
data:
  dataset: custom
  root: /kaggle/input/100-bird-species
  num_classes: 525
  size: 224
  batch_size: 64
  workers: 4
model_checkpoint:
  filename: best-step-{step}-{val_acc:.4f}
  monitor: val_acc
  save_last: true
  mode: max
```

  At about the 7th epoch, the training was abruptly stopped, it may be related to the **max_steps** parameter. So in this case, how should I set the appropriate parameters to complete the full training task ? I would appreciate it if you could give me some specific data.

**Comment by bwconrad** [comment]
If you want to train for longer, you can increase the `max_steps` argument in a config file and pass the config with `--config` or you can set it in the cli with `--trainer.max_steps`.

**Comment by hotarujun** [comment]
Thanks for your reply.

when the **max_steps** is set to 5000, the output **metrics.csv** like this:

![image](https://github.com/bwconrad/vit-finetune/assets/148742214/9c482778-0720-49d3-bc02-56bd7f7c60fc)

The train_acc and val_acc is really high. 

But when I ran evaluate command
`!python /kaggle/input/vit-finetune/main.py test --ckpt_path /kaggle/working/custom/version_0/checkpoints/'best-step-step=3500-val_acc=0.9817.ckpt' --config /kaggle/working/custom/version_0/config.yaml`

The output like this:

![image](https://github.com/bwconrad/vit-finetune/assets/148742214/83cd6d56-10d2-4987-bde1-0828270dcc86)

The test_acc is terribly low. May I ask why this happended? This is my [code](https://www.kaggle.com/code/flamedie/fine-tuning-vit-for-birds525-image-classification/notebook)

**Comment by hotarujun** [comment]
Based on the ckpt file that I have trained, and increase the max_steps to 6000, I got new test result: 

![image](https://github.com/bwconrad/vit-finetune/assets/148742214/f4d1970a-4578-4b67-a183-d1ae0e1e0218)

I'm really satisfied with the result.

Still don't know why my former test result performed so badly, but that's not important anymore.

Thanks for your amazing project ! It helps me a lot. Have a Nice day !!!
